kmod/src/net.c

```diff
@@ -948,22 +948,27 @@ static int sock_opts_and_names(struct scoutfs_net_connection *conn,
 		goto out;
 
 	addrlen = sizeof(struct sockaddr_in);
-	ret = kernel_getsockname(sock, (struct sockaddr *)&conn->sockname,
+	ret = kc_kernel_getsockname(sock, (struct sockaddr *)&conn->sockname,
 				 &addrlen);
 	if (ret == 0 && addrlen != sizeof(struct sockaddr_in))
 		ret = -EAFNOSUPPORT;
-	if (ret)
+	if (ret < 0)
 		goto out;
 
 	addrlen = sizeof(struct sockaddr_in);
-	ret = kernel_getpeername(sock, (struct sockaddr *)&conn->peername,
+	ret = kc_kernel_getpeername(sock, (struct sockaddr *)&conn->peername,
 				 &addrlen);
 	if (ret == 0 && addrlen != sizeof(struct sockaddr_in))
```

Comment:
This check is now dead.  The hope is always to have the code look like upstream and have compat to backwards in time, not the opposite.  So remove the addrlens from our calls, and have the compat wrapper for the older kernels pass in its own addrlen and return it if the kernel_ call returned 0.  Then these calls will check ret > = 0 && ret != sizeof().